Lamina/include/GL/camera.hpp

```diff
@@ -0,0 +1,145 @@
+		PivotCamera() : target(lm::vec3f({ 0, 0, 0 })), position(lm::vec3f({ 0, 0, 0 })), radius(5), pitch(0), yaw(0), window(nullptr), pitchClamp(89.f), 
+			radiusClamp(lm::vec2f({5, 50})), firstMouseMovement(true), sensitivity(0.5f), scrollSensitivity(1.0f) {};
+		PivotCamera(lm::vec3f initialPosition, lm::vec3f initialTarget, float initialRadius, lm::Window& _window) :
+			target(initialTarget), position(initialPosition), radius(initialRadius), window(&_window), pitch(0), yaw(0), pitchClamp(89.f), radiusClamp(lm::vec2f({ 5.f, 50.f })),
+		firstMouseMovement(true), sensitivity(0.5f), scrollSensitivity(1.0f)
+		{
+			glfwSetWindowUserPointer(this->window->window, this);
+			glfwSetScrollCallback(this->window->window, lm::PivotCamera::ScrollCallback);
+			glfwSetCursorPosCallback(this->window->window, lm::PivotCamera::CursorPosCallback);
+			UpdateCamera();
+		};
+		PivotCamera(lm::vec3f initialPosition, lm::vec3f initialTarget, float initialRadius, float _pitchClamp, lm::vec2f _radiusClamp, lm::Window& _window) : 
+			target(initialTarget), position(initialPosition), radius(initialRadius), window(&_window), pitch(0), yaw(0), firstMouseMovement(true), sensitivity(0.5f), scrollSensitivity(1.0f)
+		{
+			glfwSetWindowUserPointer(this->window->window, this);
+			glfwSetScrollCallback(this->window->window, lm::PivotCamera::ScrollCallback);
+			glfwSetCursorPosCallback(this->window->window, lm::PivotCamera::CursorPosCallback);
+			if (_radiusClamp.x() > 0.f && _radiusClamp.x() < _radiusClamp.y()) radiusClamp = _radiusClamp;
+			else radiusClamp = lm::vec2f({ 5.f, 50.f });
+			if (_pitchClamp > 0.f && _pitchClamp <= 90.f) pitchClamp = _pitchClamp;
+			else pitchClamp = 89.f;
+			UpdateCamera();
+		};
+
+		lm::mat4 GetCameraLookAtMatrix(){
+			lm::vec3f up = lm::normalize(lm::CrossProduct(lm::normalize(lm::CrossProduct(position, lm::vec3f({ 0, 1, 0 }))) , position));
+			return LookAtMatrix(position, target, up);
+		}
+		void UpdateCamera()
+		{
+			this->position.x() = std::cos(lm::constants::DegToRad(yaw)) * std::cos(lm::constants::DegToRad(pitch)) * radius;
+			this->position.y() = std::sin(lm::constants::DegToRad(pitch)) * radius;
+			this->position.z() = std::sin(lm::constants::DegToRad(yaw)) * std::cos(lm::constants::DegToRad(pitch)) * radius;
+		}
+		void GetScrollInput(int scroll)
+		{
+			if (scroll > 0 && this->radius >= this->radiusClamp.x()) this->radius -= 1.f * scrollSensitivity;
+			if (scroll < 0 && this->radius <= this->radiusClamp.y()) this->radius += 1.f * scrollSensitivity;
+			UpdateCamera();
+		}
+		void GetMousePosInput(lm::vec2d cursorPos)
+		{
+			if (firstMouseMovement) 
+			{
+				lastMousePosition = cursorPos;
+				firstMouseMovement = false;
+				return;
+			}
+			if (lastMousePosition.x() - cursorPos.x() > 0) 
+				yaw += ((float)lastMousePosition.x() - (float)cursorPos.x()) * sensitivity;
+			if (lastMousePosition.x() - cursorPos.x() < 0) 
+				yaw += ((float)lastMousePosition.x() - (float)cursorPos.x()) * sensitivity;
```

Comment:
This if-statement is redundant since the same thing will happen for both statements. You can combine them with an OR clause but really, all that is happening is yaw adjustment is avoided if the delta is non-zero. This is also redundant because yaw will not be changed if the delta is zero anyway. The if-statement will arguably cost more than just doing the yaw adjustment regardless of the delta.